### PR TITLE
Completed the status validation for location reports

### PR DESCRIPTION
**Pull request by hayleymoodley**
@Kenneth-Siu 
@JosephWetherlyAshcroft

### CODE REVIEW

File: AgentDiscoveries-Frontend/app/src/components/submit-forms/submit-location-report.jsx [hunk 1]
```diff
@@ -109,9 +109,20 @@ export default class LocationReportSubmit extends React.Component {
...
+            apiPost('status/reports',body)
+                .then(() => this.addMessage('Report submitted', 'info'))
+                .catch(() => this.addMessage('Error with status - make sure its less than or equal to 100', 'danger'));
+        }
+        else{
+            apiPost('reports/locationstatuses',body)
+                .then(()=> this.addMessage('Report submitted','info'))
+                .catch(()=> this.addMessage('Error submitting report, please try again later','danger'));
+        }
+
```
Comment (Kenneth-Siu) [inline AgentDiscoveries-Frontend/app/src/components/submit-forms/submit-location-report.jsx:125]: Could you double-check the white-spacing on this? Spacing might seem nit-picky but it makes it easier to read the code, and that's highly important!

**Review by Kenneth-Siu — CHANGES_REQUESTED** [review]
Remember that the ticket said

> Make sure this is validated both client-side and server-side.

At the moment, we validate it client-side (aka the front-end), but we don't validate it server-side (aka the back-end).

If you're not sure about anything, come and find me :)

File: AgentDiscoveries-Frontend/app/src/components/submit-forms/submit-location-report.jsx (hunk 1 above)
Comment (hayleymoodley) [reply AgentDiscoveries-Frontend/app/src/components/submit-forms/submit-location-report.jsx:125]: I added the backend validation and fixed the white spacing.

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/models/ReportBase.java [hunk 2]
```diff
@@ -23,11 +27,17 @@ public void setReportId(int reportId) {
...
 
+
     @Column(name = "status", nullable = false)
     public byte getStatus() {
-        return status;
+        if (status <= 100) {
+            return status;
+        } else {
+            return Byte.parseByte("INVALID");
+        }
```
Comment (Kenneth-Siu) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/models/ReportBase.java:37]: So what happens if a user manages to submit an invalid status code? Would they be allowed to do so?

It looks like they're allowed to submit things with invalid status codes and it's just that when we retrieve them we say they're invalid!

**Review by Kenneth-Siu — CHANGES_REQUESTED** [review]

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/models/ReportBase.java (hunk 2 above)
Comment (Kenneth-Siu) [reply AgentDiscoveries-Backend/src/main/java/org/softwire/training/models/ReportBase.java:37]: Sorry I didn't catch this the first time around!

Comment (hayleymoodley) [reply AgentDiscoveries-Backend/src/main/java/org/softwire/training/models/ReportBase.java:35]: I changed the backend file to return an error message

**Pull request merged**